src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -146,7 +146,7 @@ public function map(object $source, object|string|null $target = null): object
             }
         }
 
-        if (!$mappingToObject && $ctorArguments && $constructor) {
+        if (!$mappingToObject && !$map?->transform && $constructor) {
```

Comment:
This is because in case of a transform we should've already built the object is that right?